### PR TITLE
Assert task fields

### PR DESCRIPTION
**Pull request by mohsen-ebrahimi**

### CODE REVIEW

File: src/main/java/io/workflow/bpmnsimulator/fieldvalidator/AssigneeFieldValidator.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import javax.annotation.Nonnull;
+import java.util.Objects;
+
+@Slf4j
+@Service
+class AssigneeFieldValidator implements FieldValidator {
+
+    @Override
+    public FieldValidationResult validate(@Nonnull final Step step, @Nonnull final Task task) {
+        final boolean isValid = Objects.equals(task.getAssignee(), step.getAssignee());
```
Comment (rsafarpour1991) [inline src/main/java/io/workflow/bpmnsimulator/fieldvalidator/AssigneeFieldValidator.java:18]: assignee can be null
candidate user and candidate group must be check

File: src/main/java/io/workflow/bpmnsimulator/fieldvalidator/FieldValidator.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package io.workflow.bpmnsimulator.fieldvalidator;
+
+import io.workflow.bpmnsimulator.model.Step;
+import org.camunda.bpm.engine.task.Task;
+
+import javax.annotation.Nonnull;
+
+public interface FieldValidator {
```
Comment (rsafarpour1991) [inline src/main/java/io/workflow/bpmnsimulator/fieldvalidator/FieldValidator.java:8]: can be rename to Validator

File: src/main/java/io/workflow/bpmnsimulator/fieldvalidator/AssigneeFieldValidator.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import lombok.extern.slf4j.Slf4j;
+import org.camunda.bpm.engine.task.Task;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.Nonnull;
+import java.util.Objects;
+
+@Slf4j
+@Service
+class AssigneeFieldValidator implements FieldValidator {
```
Comment (rsafarpour1991) [inline src/main/java/io/workflow/bpmnsimulator/fieldvalidator/AssigneeFieldValidator.java:14]: can be rename to AssigneeValidator

File: src/main/java/io/workflow/bpmnsimulator/fieldvalidator/AssigneeFieldValidator.java (hunk 1 above)
Comment (rsafarpour1991) [inline src/main/java/io/workflow/bpmnsimulator/fieldvalidator/AssigneeFieldValidator.java:18]: task can be multi instance and assignee is collection of item

Comment (mohsen-ebrahimi) [reply src/main/java/io/workflow/bpmnsimulator/fieldvalidator/AssigneeFieldValidator.java:18]: It works fine with `null` assignee. But I've created a ticket to add a test for that: https://github.com/mohsen-ebrahimi/bpmn-simulator/issues/6

Also I've added two other tickets to support `candidateUser` and `candidateGroup`:
https://github.com/mohsen-ebrahimi/bpmn-simulator/issues/4
https://github.com/mohsen-ebrahimi/bpmn-simulator/issues/5

File: src/main/java/io/workflow/bpmnsimulator/fieldvalidator/FieldValidator.java (hunk 2 above)
Comment (mohsen-ebrahimi) [reply src/main/java/io/workflow/bpmnsimulator/fieldvalidator/FieldValidator.java:8]: Sure. It's done here:
https://github.com/mohsen-ebrahimi/bpmn-simulator/pull/3

File: src/main/java/io/workflow/bpmnsimulator/fieldvalidator/AssigneeFieldValidator.java (hunk 3 above)
Comment (mohsen-ebrahimi) [reply src/main/java/io/workflow/bpmnsimulator/fieldvalidator/AssigneeFieldValidator.java:14]: Sure. It's done here: https://github.com/mohsen-ebrahimi/bpmn-simulator/pull/3

File: src/main/java/io/workflow/bpmnsimulator/fieldvalidator/AssigneeFieldValidator.java (hunk 1 above)
Comment (mohsen-ebrahimi) [reply src/main/java/io/workflow/bpmnsimulator/fieldvalidator/AssigneeFieldValidator.java:18]: I've created a ticket for changing assignee to a collection of String: https://github.com/mohsen-ebrahimi/bpmn-simulator/issues/7
But I didn't get what you mean by `task can be multi-instance`. Let's discuss it here:

**Pull request merged**